File: base/src/main/kotlin/ch/patchcode/pr3/logbook/objects/Shiptype.kt
package ch.patchcode.pr3.logbook.objects

data class Shiptype(
		val id: Long,
		val game: Game,
		val name: String,
		val cargoSpace: Int?,
		val maneuverability: Int?,
		val draft: Int?,
		val minSpeed: Int?,
		val maxSpeed: Int?,
		val cannons: Int?,
		val sailors: Int?,
		val hitPoints: Int?,
		val dailyCost: Int?,
		val price: Int?
)

File: angular-gui/src/app/games/game/shiptypes/shiptype.ts
export class Shiptype {

  constructor(
    public id: number,
    public name: string
  ) {}

  cargoSpace: number;
  maneuverability: number;
  draft: number;
  minSpeed: number;
  maxSpeed: number;
  cannons: number;
  sailors: number;
  hitPoints: number;
  dailyCost: number;
  price: number;
}
